src/coreComponents/unitTests/xmlTests/testXMLFile.cpp

```diff
@@ -35,6 +41,197 @@ TEST( testXML, testXMLFile )
+  EXPECT_TRUE( lineFound );
+}
+/**
+ * @brief Verifies if the specified group, its children, and the associated wrappers have a
+ * DataFileContext object that correctly locate from where the objects where declared in the
+ * source file.
+ * @param document root xml document (which potentially contains includes).
+ * @param group The group to (recursively) verify.
+ * @param verifCount a set that will be filled with the names, with the form
+ * "GroupName" or "GroupName/WrapperName".
+ */
+void verifyGroupDataFileContextRecursive( xmlDocument const & document, Group const & group,
+                                          std::set< string > & verifications )
+{
+  // GEOS_LOG( "Verifying "<< group.getName());
+  if( group.getDataContext().isDataFileContext() )
+  {
+    verifyDataFileContext( dynamic_cast< DataFileContext const & >( group.getDataContext() ),
+                           document );
+    verifications.emplace( group.getName() );
+  }
+
+  for( auto const & wrapperIterator : group.wrappers() )
+  {
+    WrapperBase const * wrapper = wrapperIterator.second;
+    if( wrapper->getDataContext().isDataFileContext() )
+    {
+      verifyDataFileContext( dynamic_cast< DataFileContext const & >( wrapper->getDataContext() ),
+                             document );
+      verifications.emplace( group.getName() + '/' + wrapper->getName() );
+    }
+  }
+
+  for( auto subGroup : group.getSubGroups() )
+  {
+    verifyGroupDataFileContextRecursive( document, *subGroup.second, verifications );
+  }
+}
+
+/**
+ * @brief Returns the element that exists in setB but not in setA.
+ */
+std::set< string > getDifference( std::set< string > & setA,
+                                  std::set< string > & setB )
+{
+  std::vector< string > result( setA.size() + setB.size() );
+
+  auto it = std::set_difference( setA.begin(), setA.end(),
+                                 setB.begin(), setB.end(),
+                                 result.begin() );
+  result.resize( it - result.begin() );
+
+  return std::set< string >( result.begin(), result.end() );
+}
+
+// Tests
+// - if the line information of each nodes and attributes can be all retrieved,
+// - if the resulting Group & Wrapper hierarchy matches with the input xml documents and includes hierarchy.
+TEST( testXML, testXMLFileLines )
+{
+  xmlDocument xmlDocument;
+  ProblemManager & problemManager = getGlobalState().getProblemManager();
+
+  {
+    problemManager.parseCommandLineInput();
+    Group & commandLine = problemManager.getGroup( problemManager.groupKeys.commandLine );
+    string const & inputFileName = commandLine.getReference< string >( problemManager.viewKeys.inputFileName );
+    xmlDocument.load_file( inputFileName.c_str(), true );
+    problemManager.parseXMLDocument( xmlDocument );
+  }
+
+  GEOS_LOG( "Loaded files : " );
+  for( auto const & buffer: xmlDocument.getOriginalBuffers() )
+  {
+    GEOS_LOG( "    " << buffer.first << " (" << buffer.second.size() << " chars)" );
+  }
```

Comment:
I thought these logs could be useful for test debugging purposes, in order to manually validate the loaded files in case of a problem (especially nested inclusion issues).